vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -463,4 +470,16 @@ synchronized protected boolean hasNextContextHandler(RoutingContextImplBase cont
   synchronized protected boolean hasNextFailureHandler(RoutingContextImplBase context) {
     return context.currentRouteNextFailureHandlerIndex() < failureHandlers.size();
   }
+
+  private void handleDecodingError(RoutingContextImplBase context, Throwable e) {
+    if (router.getDecoderErrorHandler() == null)
+      context.response()
+        .putHeader(HttpHeaders.CONTENT_TYPE, "text/plain")
```

Comment:
By the way, I think content-type is not required for such responses..

Reply:
Should i send the body with info about exception?

Reply:
here's my view
- if the 400 response is to have body with the actual reason, then it shouldn't be exception message, as it's too low level. It should go like "malformed query p
string" or so..
- or we shouldn't sent any explanations to the client.
To be honest, I saw both approaches in practice :)

No matter which way you chose it would be more usufull to log that exception at least at warn level, to be able to see more detailed evidence in the log.

Reply:
Changed to a simple message and added log.warn()

Reply:
looks great. thnks